### PR TITLE
Optimize variant price and is published product filters

### PR DESCRIPTION
**Pull request by tomaszszymanski129**
I want to merge this change because it optimizes the filtering of products by "is published" and "variant price" filters.

is_published before ([450ms](https://explain.dalibo.com/plan/8a5e22cb7gd7efe7))
is_published after ([1,65ms](https://explain.dalibo.com/plan/5b2ee83779h3b5a5)) - 272x faster
variant_price before ([230ms](https://explain.dalibo.com/plan/ch8c11356e7ee3bc))
variant_price after ([3,68ms](https://explain.dalibo.com/plan/bbbcf92ed03488eb)) - 62,5x faster

# Impact

- [ ] New migrations
- [ ] New/Updated API fields or mutations
- [ ] Deprecated API fields or mutations
- [ ] Removed API types, fields, or mutations

# Docs

- [ ] Link to documentation:

# Pull Request Checklist

- [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
- [ ] Database queries are optimized and the number of queries is constant
- [ ] Database migrations are either absent or optimized for zero downtime
- [ ] The changes are covered by test cases
- [ ] All new fields/inputs/mutations have proper labels added (`ADDED_IN_X`, `PREVIEW_FEATURE`, etc.)
- [ ] All migrations have proper dependencies
- [ ] All indexes are added concurrently in migrations
- [ ] All RunSql and RunPython migrations have revert option defined

### CODE REVIEW

**Review by maarcingebala — APPROVED** [review]

File: saleor/graphql/product/filters.py [hunk 1]
```diff
@@ -511,32 +512,14 @@ def filter_collections(qs, _, value):
...
-        .values("product_id")
-    )
-
-    # Filter out product for which there is no variant with price
-    variant_channel_listings = (
-        ProductVariantChannelListing.objects.using(qs.db)
-        .filter(
-            Exists(channel.filter(pk=OuterRef("channel_id"))),
-            price_amount__isnull=False,
+    if channel := Channel.objects.using(qs.db).filter(slug=channel_slug).first():
```
Comment (fowczarek) [inline saleor/graphql/product/filters.py:515]: Are you sure that your solution will behave in the same way for not existing slug?

File: saleor/graphql/product/filters.py [hunk 2]
```diff
@@ -1246,20 +1229,16 @@ def filter_available_from(self, queryset, name, value):
...
-        )
-        variant_listing = variant_listing.values("variant_id")
-        variants = (
-            ProductVariant.objects.using(qs.db)
-            .filter(Exists(variant_listing.filter(variant_id=OuterRef("pk"))))
-            .values("product_id")
-        )
-        return qs.filter(Exists(variants.filter(product_id=OuterRef("pk"))))
+        channel = Channel.objects.using(qs.db).filter(slug=channel_slug).first()
+        if not channel:
```
Comment (fowczarek) [inline saleor/graphql/product/filters.py:1233]: Are you sure that your solution will behave in the same way for not existing slug?

**Comment by maarcingebala** [comment]
FYI - I was testing the workflow to check patches on this PR, and it failed; I checked that manually, and I think it's because this branch would need a rebase with 3.20.

**Comment by aniav** [comment]
Hey @tomaszszymanski129 what's the status of this PR? It hasn't seen any actions for some time now 🙌

**Comment by tomaszszymanski129** [comment]
@aniav there are two things I need to check here to make sure filtering behaviour is exactly as it was, though I switched to more urgent stuff and never had the time. I'll try to test it this week and if it's fine I'll merge. If there are some changes that needs to be applied it has to wait in queue. In 2nd case, I'll write an update so we don't see it's hanging open for no reason.

**Comment by tomaszszymanski129** [comment]
Outdated, closing it for now, we have a ticket for it to tackle later, it's low prio.

**Pull request closed without merging**